src/field.rs

```diff
@@ -226,6 +245,31 @@ macro_rules! make_field {
                 $fp.p as $int
             }
 
+            fn append_to(&self, bytes: &mut Vec<u8>) {
+                let int = $fp.from_elem(self.0);
+                let mut slice = [0; Self::BYTES];
+                for i in 0..Self::BYTES {
+                    slice[i] = ((int >> (i << 3)) & 0xff) as u8;
+                }
+                bytes.extend_from_slice(&slice);
+            }
+
+            fn read_from(bytes: &[u8]) -> Result<Self, FieldError> {
+                if Self::BYTES > bytes.len() {
+                    return Err(FieldError::FromBytesShortRead);
+                }
+
+                let mut int = 0;
+                for i in 0..Self::BYTES {
+                    int |= (bytes[i] as u128) << (i << 3);
+                }
+
+                if int >= $fp.p {
+                    return Err(FieldError::FromBytesModulusOverflow);
+                }
+                Ok(Self($fp.elem(int)))
+            }
```

Comment:
It bugs me slightly that these methods aren't symmetric. It also seems unfortunate that `append_to` is only capable of appending to a `Vec<u8>`. It would be more generally useful to provide routines that allow transforming a `FieldElement` to and from `[u8]`, and then you could do anything you want with the byte array, whether it's appending to a `Vec` or [writing to an `std::io::Write`](https://doc.rust-lang.org/std/io/trait.Write.html#tymethod.write).

I wonder: what if instead of these two methods on `FieldElement`, you had `FieldElement` require `TryFrom<[u8]>`, and provided `From $elem for [u8]`?

Reply:
As discussed on the design call, @acmiyaguchi will look into replacing these functions with implementations of whatever traits are needed by serde. See https://github.com/abetterinternet/libprio-rs/issues/24. I've added TODOs in the code to remind us.